networkx/utils/backends.py

```diff
@@ -126,6 +138,12 @@ def _get_backends(group, *, load_and_call=False):
 
 backends = _get_backends("networkx.backends")
 backend_info = _get_backends("networkx.backend_info", load_and_call=True)
+# Initialize default configuration from backends (if provided)
+backend_config.update(
+    (backend, info["default_config"])
+    for backend, info in backend_info.items()
+    if "default_config" in info
+)
```

Comment:
here, it might be more optimal to have a separate function(like `get_info`) that returns only the backend's configurations, like `get_config(config_dict=None)`(if `config_dict` is not `None` then the configuration dictionary is updated and returned otherwise the default configurations are returned) because config dictionary would probably be updated quite frequently and we probably shouldn't load all the `backend_info` every time. Please let me know if I'm missing something here.

Reply:
I'm not sure I completely understand your comment. `get_info` is called exactly once at import time for each backend (before this PR and in this PR). Among other things, this helps the dispatch machinery know which backends implement which functions.

By optionally including `"default_config"` in the dict returned by `get_info`, we are able to initialize the _default_ configuration for each backend at import time. This happens only once. This may be nice for users so they can do things like:
```python
import networkx as nx
nx.backend_config["parallel"]["n_job"] = 5
```
`nx.backend_config` is how we expect users to update configuration.

We could make a separate entry point for the default config, but I don't know if it would be worth doing since it seems to me that the "backend_info" entry point works well enough.

Reply:
ok, I thought this kind of thing was possible :
```.py
import networkx as nx
nx.backend_config["parallel"]["n_job"] = 5
# defining G
print(nx.betweeness_centrality(G, backend="parallel"))

nx.backend_config["parallel"]["n_job"] = -1
nx.backend_config["parallel"]["backend"] = 'threading'
print(nx.betweeness_centrality(G, backend="parallel"))
```
so I thought we would be extracting all the configurations and adding them in the `joblib.Parallel()` in all the functions, so having a separate `get_config` function made sense to me.

But, I think, based on the clarification above, that all the configuration code lines will be executed while importing networkx so the final configuration for both the function calls would be `{"n_jobs" : -1, "backend" : "threading"}`, right?

Reply:
Coming back to this question:
- I think the code you write above *would* work with this configuration system.
- the default values will be obtained from the backend_info `get_info` function.
- the user overrides those values as desired for each function call.
- the function dispatches to nx-parallel which looks at the config to get the values to send to joblib. 

I think your `get_config` function would live in nx-parallel and look at `nx.config.backends.parallel.<name>` for each option. Or, we could use the whole config object like: `joblib.Parallel(**nx.config.backends.parallel)`. But the user would be able to change them between each function call.